src/lib/message/L1ToL2Message.ts

```diff
@@ -245,17 +246,50 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     const creationReceipt =
       retryableCreationReceipt ||
       (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
-
-    if (creationReceipt) {
-      const l2Receipt = new L2TransactionReceipt(creationReceipt)
-      const redeemEvents = l2Receipt.getRedeemScheduledEvents()
+    if (creationReceipt){
+      const creationBlockNumber = creationReceipt.blockNumber
+      const ticketId = creationReceipt.transactionHash
+      const iFace = new Interface([
+        'event RedeemScheduled(     bytes32 indexed ticketId,     bytes32 indexed retryTxHash,     uint64 indexed sequenceNum,     uint64 donatedGas,     address gasDonor )',
+      ])
+      const redeemTopic = iFace.getEventTopic('RedeemScheduled')
+
+      let redeemEventLogs: ethers.providers.Log[] = []
+      let fromBlockNumber: number = creationBlockNumber
+      let fromBlock = await this.l2Provider.getBlock(fromBlockNumber)
+      let increment: number = 1000
+      const creationBlock = await this.l2Provider.getBlock(creationBlockNumber)
+      const maxBlock = await this.l2Provider.getBlockNumber()
+      while(fromBlockNumber <= maxBlock) {
+        const toBlockNumber = Math.min(fromBlockNumber + increment, maxBlock)
+        redeemEventLogs = await this.l2Provider.getLogs( {fromBlock: fromBlockNumber, toBlock: toBlockNumber, topics: [redeemTopic, ticketId]})
+        if (redeemEventLogs.length != 0) break
+        const toBlock = await this.l2Provider.getBlock(toBlockNumber)
+        if ((toBlock.timestamp - creationBlock.timestamp) > RETRYABLELIFETIMESECONDS) break
+        const processedSeconds = (toBlock.timestamp - fromBlock.timestamp)
+        if (processedSeconds != 0) {
+          // find the increment that cover ~ 1 day
+          increment = increment * RETRYABLELIFETIMESECONDS / 7 / processedSeconds
```

Comment:
Sanity check: so iterations of `increment` would go:
1. 1000 blocks
2. ~ # of blocks in 1 day
3. ~ # of blocks in 1 day
4. ... etc.

Reply:
unless toBlock.timestamp == fromBlock.timestamp, then it would keep requesting 1000 blocks
I think we can change the `1000` constant if we have an estimate of how much block to expect between a timestamp update